updatecli/policies/open-release-pr/updatecli.d/open-release-pr.yaml

```diff
@@ -0,0 +1,164 @@
+name: Open a PR that prepares the policy for release
+pipelineid: "{{ .pipelineid }}"
+
+sources:
+  newestGitHubDraftReleaseVersion:
+    name: Extract version from newest GitHub draft release from release drafter
+    kind: githubrelease
+    spec:
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.repo }}"
+      token: "{{ requiredEnv .github.token }}"
+      key: "title"
+      typefilter:
+        draft: true
+      versionfilter:
+        kind: latest
+    transformers:
+      - trimprefix: "v"
+
+  newestReleaseTagVersion:
+    name: Extract newest tag from newest GitHub release
+    # used for displaying a gh compare URL
+    # disablesourceinput: true # we continue on fail (we may not have a tag)
+    kind: githubrelease
+    spec:
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.repo }}"
+      token: "{{ requiredEnv .github.token }}"
+      key: "title"
+      versionfilter:
+        kind: latest
+
+  cargoPackageName:
+    name: Extract cargo package name from Cargo.toml
+    kind: toml
+    scmid: "policy-repo"
+    dependson:
+      - condition#checkCargoTomlExists
+    spec:
+      file: "Cargo.toml"
+      key: "package.name"
+
+conditions:
+  checkCargoTomlExists:
+    name: Check if Cargo.toml exists
+    scmid: "policy-repo"
+    kind: file
+    disablesourceinput: true
+    spec:
+      file: Cargo.toml
+  checkCargoLockExists:
+    name: Check if Cargo.lock exists
+    scmid: "policy-repo"
+    kind: file
+    disablesourceinput: true
+    spec:
+      file: Cargo.lock
+
+targets:
+  updateMetadataAnnotationVersion:
+    name: update metadata.yml with new version
+    kind: yaml
+    scmid: "policy-repo"
+    sourceid: newestGitHubDraftReleaseVersion
+    spec:
+      searchpattern: true
+      file: "metadata.y*ml"
+      key: "$.annotations.'io.kubewarden.policy.version'"
+      # value: '{{ source  "newestGitHubDraftReleaseVersion" }}'
+
+  updateCargoTomlVersion:
+    name: update Cargo.toml with new version
+    scmid: "policy-repo"
+    disablesourceinput: true
+    dependson:
+      - condition#checkCargoTomlExists
+    kind: shell
+    spec:
+      shell: /usr/bin/bash
+      command: |
+        export new_version='{{ source "newestGitHubDraftReleaseVersion" }}'
+        awk -v new_version="$new_version" '
+          BEGIN { in_package = 0 }
+          /^\[package\]/ { in_package = 1 }
+          in_package && /^version = / { sub(/version = ".*"/, "version = \"" new_version "\"") }
+          /^\[/ && !/^\[package\]/ { in_package = 0 }
+          { print }
+        ' Cargo.toml > Cargo.toml.tmp && mv Cargo.toml.tmp Cargo.toml
```

Comment:
Can't we update the file using the [built-in support of toml](https://www.updatecli.io/docs/plugins/resource/toml/) that updatecli provides?

In a similar way to how we update the `metadata.yml`

Reply:
oh my... I now understand

However, can't we achieve the same thing with a simpler `sed` invocation like this one:

```console
sed -i 's/^version = ".*"/version = "6.6.6"/' Cargo.toml
```

The package version is always on it's own dedicated line, starting from the beginning of the line. There's no risk (IMHO) of having this expression replace a line like `crate-foo = { version = '1.2.3' }`

Reply:
Yes the library that I am using on Updatecli don't keep the initial format of the file and I am not aware of another one that would work better

Reply:
By the way, Updatecli is using dasel as a library to query toml files

Reply:
btw would it make sense to use another tool to format the file?

Reply:
then we could use the toml plugin in Updatecli which I find easier to read and maintain and then run another tool if available

Reply:
> By the way, Updatecli is using dasel as a library to query toml file

I remember needing have the cli installed locally for it to work, but looking at updatecli's code, indeed.

> btw would it make sense to use another tool to format the file?

I'm ok with awk for now tbh, I'm more concerned on getting the targets to not fail if the conditions don't pass.

Reply:
I feel so stupid... it's just a matter of running:

```console
cargo set-version 1.2.3
```

This will bump the version inside of the `Cargo.toml` file.

After that, we can either run `cargo check` (which will be a bit slower and could fail due to new linters) or we could run `cargo fetch`

Reply:
JFYI: this is part of https://crates.io/crates/cargo-edit

Reply:
cargo-edit sounds the most robust solution

Reply:
Done in d4a7728.